### ISSUE TITLE
SyncMultiGPUTrainerReplicated  ProgressBar(extra_display)

### ISSUE THREAD

**Issue by adeagle** [ppwwyyxx/tensorpack#442]
when train with  SyncMultiGPUTrainerReplicated

extra_display = ["cost"]

 extra_callbacks=[
           ProgressBar(extra_display),     
 ],  
Failed with:
"The name 'cost:0' refers to a Tensor which does not exist. The operation, 'cost', does not exist in the graph."

but train with SimpleTrainer  is normal?

how can i get the loss when train with SyncMultiGPUTrainerReplicated

**Comment by ppwwyyxx** [comment]
With replicated training, the tensor name becomes `tower0/cost`. Could you try with that?

**Comment by adeagle** [comment]
thanks,problem solved.
more question:

i want to train with 4 gpus use Between-graph mode,how to do?

**Comment by ppwwyyxx** [comment]
Could you explain what is "Between-graph mode". Are you still talking about multi-gpu training or distributed training?

**Comment by adeagle** [comment]
multi-gpu training,beacuase io overload

**Comment by adeagle** [comment]
time python tf_cnn_benchmarks.py --local_parameter_device=cpu --num_gpus=2 --num_intra_threads=7 --num_inter_threads=7 \
--batch_size=32 --model=vgg16 --variable_update=parameter_server
TensorFlow:  1.4
Model:       vgg16
Mode:        training
SingleSess:  False
Batch size:  64 global
             32 per device
Devices:     ['/gpu:0', '/gpu:1']
Data format: NCHW
Optimizer:   sgd
Variables:   parameter_server
==========
Generating model
Running warm up
Done warm up
Step	Img/sec	loss
1	images/sec: 147.1 +/- 0.0 (jitter = 0.0)	7.255
10	images/sec: 146.6 +/- 0.3 (jitter = 0.6)	7.255
20	images/sec: 146.8 +/- 0.2 (jitter = 0.7)	7.255
30	images/sec: 146.9 +/- 0.1 (jitter = 0.5)	7.255
40	images/sec: 146.8 +/- 0.1 (jitter = 0.7)	7.255
50	images/sec: 146.7 +/- 0.1 (jitter = 0.7)	7.255
60	images/sec: 146.7 +/- 0.1 (jitter = 0.8)	7.255
70	images/sec: 146.6 +/- 0.1 (jitter = 0.9)	7.255
80	images/sec: 146.7 +/- 0.1 (jitter = 0.9)	7.255
90	images/sec: 146.6 +/- 0.1 (jitter = 0.9)	7.255
100	images/sec: 146.6 +/- 0.1 (jitter = 0.9)	7.255
----------------------------------------------------------------
total images/sec: 146.38
----------------------------------------------------------------

real	0m59.740s
user	2m59.392s
sys	0m5.960s

time python tf_cnn_benchmarks.py --local_parameter_device=cpu --num_gpus=1 --num_intra_threads=7 --num_inter_threads=7 \
--batch_size=64 --model=vgg16 --variable_update=parameter_server 
TensorFlow:  1.4
Model:       vgg16
Mode:        training
SingleSess:  False
Batch size:  64 global
             64 per device
Devices:     ['/gpu:0']
Data format: NCHW
Optimizer:   sgd
Variables:   parameter_server
==========
Generating model
Running warm up
Done warm up
Step	Img/sec	loss
1	images/sec: 83.9 +/- 0.0 (jitter = 0.0)	7.255
10	images/sec: 81.3 +/- 0.6 (jitter = 1.9)	7.255
20	images/sec: 81.5 +/- 0.4 (jitter = 2.1)	7.255
30	images/sec: 81.8 +/- 0.3 (jitter = 2.2)	7.255
40	images/sec: 81.9 +/- 0.3 (jitter = 1.8)	7.255
50	images/sec: 81.9 +/- 0.2 (jitter = 1.9)	7.255
60	images/sec: 81.8 +/- 0.2 (jitter = 2.0)	7.255
70	images/sec: 81.8 +/- 0.2 (jitter = 2.0)	7.255
80	images/sec: 82.0 +/- 0.2 (jitter = 1.8)	7.255
90	images/sec: 82.0 +/- 0.2 (jitter = 1.9)	7.255
100	images/sec: 82.1 +/- 0.2 (jitter = 1.8)	7.255
----------------------------------------------------------------
total images/sec: 82.04
----------------------------------------------------------------

real	1m37.090s
user	1m27.708s
sys	0m6.428s

time python tf_cnn_benchmarks.py --local_parameter_device=cpu --num_gpus=2 --num_intra_threads=7 --num_inter_threads=7 \
--batch_size=32 --model=vgg16 --variable_update=replicated --use_nccl=True
TensorFlow:  1.4
Model:       vgg16
Mode:        training
SingleSess:  False
Batch size:  64 global
             32 per device
Devices:     ['/gpu:0', '/gpu:1']
Data format: NCHW
Optimizer:   sgd
Variables:   replicated
AllReduce:   nccl
==========
Generating model
Running warm up
Done warm up
Step	Img/sec	loss
1	images/sec: 114.5 +/- 0.0 (jitter = 0.0)	7.255
10	images/sec: 115.2 +/- 0.4 (jitter = 0.8)	7.255
20	images/sec: 115.1 +/- 0.3 (jitter = 1.0)	7.255
30	images/sec: 115.1 +/- 0.2 (jitter = 1.0)	7.255
40	images/sec: 115.0 +/- 0.2 (jitter = 1.4)	7.255
50	images/sec: 114.9 +/- 0.2 (jitter = 1.3)	7.255
60	images/sec: 114.9 +/- 0.2 (jitter = 1.2)	7.255
70	images/sec: 114.9 +/- 0.2 (jitter = 1.3)	7.255
80	images/sec: 114.9 +/- 0.1 (jitter = 1.2)	7.255
90	images/sec: 114.9 +/- 0.1 (jitter = 1.3)	7.255
100	images/sec: 114.8 +/- 0.1 (jitter = 1.3)	7.255
----------------------------------------------------------------
total images/sec: 114.40
----------------------------------------------------------------

real	1m13.169s
user	0m45.404s
sys	0m5.104s


time python tf_cnn_benchmarks.py --local_parameter_device=cpu --num_gpus=1 --num_intra_threads=7 --num_inter_threads=7 \
--batch_size=64 --model=vgg16 --variable_update=replicated --use_nccl=True
TensorFlow:  1.4
Model:       vgg16
Mode:        training
SingleSess:  False
Batch size:  64 global
             64 per device
Devices:     ['/gpu:0']
Data format: NCHW
Optimizer:   sgd
Variables:   replicated
AllReduce:   nccl
==========
Generating model
Running warm up
Done warm up
Step	Img/sec	loss
1	images/sec: 92.8 +/- 0.0 (jitter = 0.0)	7.255
10	images/sec: 93.0 +/- 0.1 (jitter = 0.4)	7.255
20	images/sec: 92.8 +/- 0.1 (jitter = 0.4)	7.255
30	images/sec: 92.7 +/- 0.1 (jitter = 0.3)	7.255
40	images/sec: 92.7 +/- 0.1 (jitter = 0.3)	7.255
50	images/sec: 92.7 +/- 0.1 (jitter = 0.3)	7.255
60	images/sec: 92.5 +/- 0.1 (jitter = 0.4)	7.255
70	images/sec: 92.5 +/- 0.1 (jitter = 0.4)	7.255
80	images/sec: 92.5 +/- 0.1 (jitter = 0.3)	7.255
90	images/sec: 92.5 +/- 0.1 (jitter = 0.3)	7.255
100	images/sec: 92.5 +/- 0.1 (jitter = 0.3)	7.255
----------------------------------------------------------------
total images/sec: 92.60
----------------------------------------------------------------

real	1m26.327s
user	0m27.088s
sys	0m5.172s


I want to add more gpu to speed up train,but  after test ,the output show the efficient is low.

**Comment by ppwwyyxx** [comment]
This issue isn't about tensorpack. You should report to tensorflow/benchmarks.

**Comment by adeagle** [comment]
thank you

**Comment by ppwwyyxx** [comment]
About the original question, after latest changes 897d29e3f1dbc5 to the trainer, using "cost" would be sufficient. "tower0/cost" also works.

**Comment by adeagle** [comment]
ok,thanks
